src/components/templates/ProtectedPageTemplate.js

```diff
@@ -0,0 +1,79 @@
+import React, { useState, useEffect } from "react";
+import styled from "styled-components";
+import PageTemplate from "./PageTemplate";
+import UserService from "../../services/UserService";
+import Link from "next/link";
+
+const Wrapper = styled("section")`
+	width: 600px;
+	margin: 0 auto;
+`;
+
+const Links = styled("ul")`
+	display: flex;
+	justify-content: space-evenly;
+	list-style: none;
+	
+	li { display: inline }
+`;
+
+const PageState = {
+	LOADING: 0,
+	NO_PERMISSION: 1,
+	HAS_PERMISSION: 2
+};
+
+function ProtectedPageTemplate({ children, page, meta }) {
+	const [visible, setVisible] = useState(PageState.HAS_PERMISSION);
+	const users = new UserService();
+
+	// useEffect(() => {
+	// 	(async () => {
+	// 		try {
+	// 			await users.getCurrentUser();
+	//
+	// 			setVisible(PageState.HAS_PERMISSION);
+	// 		} catch (error) {
+	// 			setVisible(PageState.NO_PERMISSION);
+	// 		}
+	// 	})();
+	// }, []);
```

Comment:
@Jason2605 @AtrumAmantus as mentioned in Discord, this is where I believe the authentication issue resides. Commenting out this block presents no authentication issues. However, we want to check that the user is logged in before showing the page - any ideas?